helix-term/src/commands.rs

```diff
@@ -1203,15 +1202,35 @@ mod cmd {
 
     fn write_quit(cx: &mut compositor::Context, args: &[&str], event: PromptEvent) {
         let (view, doc) = current!(cx.editor);
-        if let Err(e) = write_impl(view, doc, args.first()) {
-            cx.editor.set_error(e.to_string());
-            return;
+        match write_impl(view, doc, args.first()) {
+            Ok(handle) => {
+                if let Err(e) = helix_lsp::block_on(handle) {
+                    cx.editor.set_error(e.to_string());
+                    return;
+                }
+            }
+            Err(e) => {
+                cx.editor.set_error(e.to_string());
```

Comment:
Can we even see the error when we already quit?

Reply:
Ah, nevermind. Yeah, it returns here.

Reply:
It should return an error before it quits.